dokka-integration-tests/gradle/build.gradle.kts

```diff
@@ -215,5 +219,28 @@ testing {
                 systemProperty("junit.jupiter.execution.parallel.mode.default", "CONCURRENT")
             }
         }
+
+        // Register specific Gradle test runs for each example so the examples can be tested individually.
```

Comment:
nit: we discussed this in the previous PR, but I still feel that it would be more conventional to have a generated `testGen` source directory (under git and with an idempotency check) than a per test project Gradle task

Reply:
One of the benefits is that you could then run an arbitrary subset of such tests using the standard `Test` task's `--tests` filter

Reply:
Yes, those would be good benefits. I'm just struggling to think about how to generate tests dynamically though, because I usually find code generation inconvenient and prone to mistakes.

Reply:
If you decide to look further into the direction of generating test cases, I think the setup is usually something like:
* `src/MyTestsInfrastructure.kt`
```kotlin
abstract class TestsInfrastructure  {
  // hand craft some eloquent testing infrastructure with an API that easy to generate in 1 LOC
  fun runTest(testDataPath: String) { ... }
}
```
* `src/tests-gen/Tests.kt`
```kotlin
// This class is generated
class Tests : TestsInfrastructure {
  @Test
  fun test_basicExample() = runTest("path/to/basicExample")
}
```